### PR TITLE
LOG-4812: Refactor metrics templates and helper functions

### PR DESCRIPTION
**Pull request by jcantrill**
### Description
This PR:
* Moves metrics and throttle templates to separate files
* Moves metrics templates to output package since that is what it is
* Refactors TLS Options logic into separate helper function
* Refactors Secret retrieval into helper function

### Links
* https://issues.redhat.com/browse/LOG-4812
* Blocks [LOG-4743](https://issues.redhat.com//browse/LOG-4743)

cc @syedriko @cahartma

### CODE REVIEW

File: internal/generator/vector/helpers/tls_options.go [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+func SetTLSProfileOptions(o logging.OutputSpec, op framework.Options) {
+	if o.Name == logging.OutputNameDefault && o.Type == logging.OutputTypeElasticsearch {
+		op[framework.MinTLSVersion] = ""
+		op[framework.Ciphers] = ""
+	} else {
+		outMinTlsVersion, outCiphers := op.TLSProfileInfo(o, ",")
+		op[framework.MinTLSVersion] = outMinTlsVersion
+		op[framework.Ciphers] = outCiphers
+	}
+}
```
Comment (syedriko) [inline internal/generator/vector/helpers/tls_options.go:17]: How 'bout
```
func SetTLSProfileOptions(o logging.OutputSpec, op framework.Options) {
    op[framework.MinTLSVersion], op[framework.Ciphers] = func()(String, String) {
        if o.Name == logging.OutputNameDefault && o.Type == logging.OutputTypeElasticsearch {
            return "", ""
        } else {
            return op.TLSProfileInfo(o, ",")
        }
    }()
}
```

File: internal/generator/vector/output/common/throttle.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import (
+	"fmt"
+	logging "github.com/openshift/cluster-logging-operator/apis/logging/v1"
+	"github.com/openshift/cluster-logging-operator/internal/generator/framework"
+	"github.com/openshift/cluster-logging-operator/internal/generator/vector/helpers"
+	"github.com/openshift/cluster-logging-operator/internal/generator/vector/normalize"
+)
+
+var (
+	SinkTransformThrottle = "sink_throttle"
```
Comment (syedriko) [inline internal/generator/vector/output/common/throttle.go:12]: Not used outside of package

File: internal/generator/vector/output/common/throttle.go [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+	UserDefinedSinkThrottle = fmt.Sprintf(`%s_%%s`, SinkTransformThrottle)
+)
+
+func AddThrottleForSink(spec *logging.OutputSpec, inputs []string) []framework.Element {
+	id := fmt.Sprintf(UserDefinedSinkThrottle, spec.Name)
+	return NewThrottleForSink(id, spec, inputs)
+}
+
+func NewThrottleForSink(id string, spec *logging.OutputSpec, inputs []string) []framework.Element {
```
Comment (syedriko) [inline internal/generator/vector/output/common/throttle.go:22]: Not used outside of package, too

File: internal/generator/vector/source/metrics.go [hunk 4]
```diff
@@ -1,10 +1,9 @@
 package source
 
 const (
-	InternalMetricsSourceName        = "internal_metrics"
-	PrometheusOutputSinkName         = "prometheus_output"
-	PrometheusExporterListenPort     = `24231`
-	AddNodenameToMetricTransformName = "add_nodename_to_metric"
+	InternalMetricsSourceName    = "internal_metrics"
+	PrometheusOutputSinkName     = "prometheus_output"
```
Comment (syedriko) [inline internal/generator/vector/source/metrics.go:5]: these two should've gone to the metrics package it seems

File: internal/generator/vector/output/common/throttle.go (hunk 2 above)
Comment (jcantrill) [reply internal/generator/vector/output/common/throttle.go:12]: Updated code to remove vars

File: internal/generator/vector/output/common/throttle.go (hunk 3 above)
Comment (jcantrill) [reply internal/generator/vector/output/common/throttle.go:22]: Moved to 'normalize' and made more generic

File: internal/generator/vector/input/inputs.go [hunk 5]
```diff
@@ -46,7 +45,7 @@ var (
 	AddLogTypeAudit = fmt.Sprintf(".log_type = %q", logging.InputNameAudit)
 
 	UserDefinedInput          = fmt.Sprintf("%s.%%s", RouteApplicationLogs)
-	UserDefinedSourceThrottle = fmt.Sprintf("%s_%%s", SourceTransformThrottle)
+	UserDefinedSourceThrottle = `source_throttle_%s`
```
Comment (syedriko) [inline internal/generator/vector/input/inputs.go:48]: UserDefinedSourceThrottle and perContainerLimitKeyField look like consts

File: internal/generator/vector/output/metrics/prometheus.go [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+[sinks.{{.ID}}.tls]
+enabled = true
+key_file = "/etc/collector/metrics/tls.key"
+crt_file = "/etc/collector/metrics/tls.crt"
+min_tls_version = "{{.TlsMinVersion}}"
+ciphersuites = "{{.CipherSuites}}"
+{{end}}`
+}
+
+type AddNodenameToMetric struct {
```
Comment (syedriko) [inline internal/generator/vector/output/metrics/prometheus.go:42]: AddNodeNameToMetric is only used in one place, together with the PrometheusOutput, may be we could just inline it into `func (p PrometheusExporter) Template() string {` and remove all the  AddNodenameToMetric code?

Comment (syedriko) [reply internal/generator/vector/output/metrics/prometheus.go:42]: And even merge `internal_metrics` into that same template ^^^?

Comment (jcantrill) [reply internal/generator/vector/output/metrics/prometheus.go:42]: sounds like a future refactoring

**Pull request merged**